### PR TITLE
Fix waypoints next and previous when connecting two lines with opposite directions

### PR DESCRIPTION
**Pull request by matejm42**
#### Description

Fixing issue #7136 - detailed description there

Fix waypoints next and previous when connecting two lines with opposite directions.

Probably not the best solution, but it works for our case. The waypoint that would cause conflict when calling next from it is not returned.

#### Where has this been tested?

  * **Platform(s):** Ubuntu 20.24
  * **Python version(s):** 3.8 and 3.11.6
  * **Unreal Engine version(s):** 4.26

#### Possible Drawbacks

Loading maps and getting waypoints from long distances can be a bit slower. However I did not experience any decrease in performance in my case. 

You can not get the next waypoint, if calling next from that waypoint would cause a loop. (We found this situation only on curbs connecting two lanes in opposite directions. In this case, it makes sense for us to behave as if the curbs are not connected together. Hopefully, there are no opposite-direction driving lanes connected together. 

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7137)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+  * Fixed waypoint.next and previous causing loops when connecting two opposite-direction lanes.
```
Comment (wentasah) [inline CHANGELOG.md:1]: Isn't this better?
```suggestion
  * Fixed waypoint.next and .previous causing loops when two opposite-direction lanes follow each other in the map.
```

**Comment by Blyron** [comment]
Please point this to UE4_DEV branch pls

**Comment by matejm42** [comment]
> Please point this to UE4_DEV branch pls

Ah mb; changed

**Comment by Blyron** [comment]
Could you resolve CHANGELOG.md conflicts?

**Review by Blyron — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7137)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @matejm42)

**Review by Blyron — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7137)* status: all files reviewed, 1 unresolved discussion (waiting on @matejm42)

**Pull request merged**